lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,61 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
+    require 'win32/registry'
+
+    # Ensure the symlink folder exists and is pointing to the proper location.
+    # This also ensures the directory structure is in place prior to creating
+    # the symlink, otherwise the symlink creation fails.
+    def set_symlink_folder
+      puppet_modules_vendor_folder = "#{get_program_files_dir}\\WindowsPowerShell\\Modules\\PuppetVendoredModules"
+      recreate_link = false
+      if Puppet::FileSystem.exist?(puppet_modules_vendor_folder) || Puppet::FileSystem.symlink?(puppet_modules_vendor_folder)
+        if !Puppet::FileSystem.symlink?(puppet_modules_vendor_folder)
+          recreate_link = true
+        else
+          recreate_link = true if Puppet::FileSystem.readlink(puppet_modules_vendor_folder) != vendored_modules_path
+          Puppet::FileSystem.unlink(puppet_modules_vendor_folder) if recreate_link
+        end
+      end
+      if recreate_link || !Puppet::FileSystem.exist?(puppet_modules_vendor_folder) || !Puppet::FileSystem.symlink?(puppet_modules_vendor_folder)
+        Puppet.debug "Creating symlink at '#{puppet_modules_vendor_folder}' \n pointed to '#{vendored_modules_path}'."
+        Puppet::FileSystem.dir_mkpath(puppet_modules_vendor_folder) if !Puppet::FileSystem.dir_exist?(puppet_modules_vendor_folder)
+        Puppet::FileSystem.symlink(vendored_modules_path, puppet_modules_vendor_folder,{:force => true})
```

Comment:
I should have read the docs a little better. If it's a directory it doesn't do anything. https://github.com/puppetlabs/puppet/blob/master/lib/puppet/file_system.rb#L230-L235